elro/mqtt.py

```diff
@@ -113,12 +113,55 @@ async def handle_device_discovery(self, device):
                 QOS_1
             )
 
+    async def device_message_task(self, hub):
+        """
+        The main loop for handling messages that will be sent to the device
+        :param device: The device that will get the message
+        """
+        while True:
+            await self.handle_device_messages(hub)
+
+    async def handle_device_messages(self, hub):
+        async with open_mqttclient(uri=self.broker_host) as client:
+            logging.info(f"Subscribing to topic 'f{self.base_topic}/elro/[device_id]]/set/[command_topic]'")
+            async with client.subscription(f"{self.base_topic}/elro/+/set/+", codec="utf8") as subscription:
+                async for msg in subscription:
+                    mqtt_message = msg.data.strip('\"')
+                    logging.info(f"Got message '{mqtt_message}' on topic '{msg.topic}'")
+                    topic_items = msg.topic.split('/')
+                    device_index = None
+                    topic_item = None
+                    for i in range(len(topic_items)): #searching for the device index and the command
+                        if topic_items[i].lower() == 'elro':
+                            try:
+                                device_index = int(topic_items[i+1])
+                                topic_item = str(topic_items[i+3])
+                            except:
```

Comment:
Can you specify the exceptions you want to capture here? Capturing all tends to eat useful information.

Reply:
Do you mean like this `except KeyError:`, or do you want details like incorrect device id and/or incorrect topic item as error? Personally if there should be reporting i thing this would be best
```Python
except as Exception as Error:
     logging.error("something about id as integer and topic item should exist")
```

I do like error reporting, i am not very good in capturing every error seperatly so i like to have a generic error

Reply:
I meant the former. I think an `except Exception:` block this diminishes the error reporting. If an exception happens, that we didn't think of, the app should actually crash and print the stack trace in my opinion, so I would vote for removing the catch all block. (And the other in this file).

Reply:
in this case i am not sure if i agree on that one. this is code for state changing, and because the main goal of elro connects is to report about the sensors, i personally would rather not have that the program crashing due to some state changing. i would rather have it always report the devices and that it is very robust in regards of that.

my solution to know what is happening and keep everything up and running would be like this
```python
except Exception as error:
     logging.error(f"something about id as integer and topic item should exist. the error reported: '{error}'")
```

In this case the error is reported and could be logged to disk. therefore it can be traced back, but the program is still doing its thing.

Reply:
@hildensia is this wat you meant? Or do you want the application to also raise an error?
```Python
except KeyError:
    logging.error("Please provide the topic as [base_topic]/elro/[device_id]/[topic_name]")
except ValueError:
    logging.error("Please provide an integer for the device_index and/or a valid topic name")
except Exception as error:
    logging.error(f"Unknown error occured '{error}'")
```

Reply:
@hildensia can you please respond to the question above? I got another PR that i want to publish 😊